integration/yarn/src/test/java/alluxio/yarn/ClientTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.yarn;
+
+import org.apache.hadoop.yarn.api.protocolrecords.GetNewApplicationResponse;
+import org.apache.hadoop.yarn.api.records.Resource;
+import org.apache.hadoop.yarn.client.api.YarnClient;
+import org.apache.hadoop.yarn.client.api.YarnClientApplication;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.mockito.Mockito.*;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.PropertyKey;
+import alluxio.exception.ExceptionMessage;
+
+
+/**
+ * Unit tests for {@link Client}.
+ * TODO: ALLUXIO-1503: add more unit test for alluxio.yarn.Client
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Client.class, YarnClient.class})
+public final class ClientTest {
+
+  private YarnClient mYarnClient;
+
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+
+  @Before
+  public void before() {
+    // Mock Yarn client
+    PowerMockito.mockStatic(YarnClient.class);
+    mYarnClient = (YarnClient) Mockito.mock(YarnClient.class);
+    Mockito.when(YarnClient.createYarnClient()).thenReturn(mYarnClient);
+  }
+
+  /**
+   * Tests that the main method properly reads args.
+   */
+  @Test(timeout = 10000)
+  public void mainTest() throws Exception {
```

Comment:
This test currently checks only that (1) the `main` method calls `new Client()` and (2) the `run` method is called on the created `Client`. We should only test the public API and not make assumptions about the internals of methods, so if we really wanted to check that args are properly processed, we would need to check this by observing how the `Client` interacts with the `YarnClient/` and `YarnClientApplication`.

I think in this case it makes sense to remove this test for now and then properly test args when ALLUXIO-1503 is addressed.